qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -313,6 +322,217 @@ def _run(self, qobj, job_id=''):
+            if isinstance(inst, (WhileLoopOp, ForLoopOp, IfElseOp, BreakLoopOp, ContinueLoopOp)):
+                return True
+        return False
+
+    def _decompose_subcircuits(self, circuit, basis_gates=None):
+        if circuit is None:
+            return None
+        if basis_gates is None:
+            basis_gates = self.configuration().basis_gates + ['for_loop', 'while_loop',
+                                                              'if_else', 'break_loop',
+                                                              'continue_loop']
+        circuit = circuit.copy()
+        for inst, _, _ in circuit.data:
+            if isinstance(inst, WhileLoopOp):
+                body, = inst.params
+                inst.params = (self._decompose_subcircuits(body, basis_gates),)
+            elif isinstance(inst, ForLoopOp):
+                loop_parameter, indexset, body = inst.params
+                inst.params = (loop_parameter, indexset,
+                               self._decompose_subcircuits(body, basis_gates))
+            elif isinstance(inst, IfElseOp):
+                true_body, false_body = inst.params
+                inst.params = (self._decompose_subcircuits(true_body, basis_gates),
+                               self._decompose_subcircuits(false_body, basis_gates))
+        circuit = transpile(circuit, basis_gates=basis_gates, optimization_level=0)
+        return circuit
+
+    def _inline_for_loop_op(self, inst):
+        indexset, loop_parameter, body = inst.params
+
+        self._last_flow_id += 1
+        loop_id = self._last_flow_id
+        loop_name = f'loop_{loop_id}'
+
+        ret = QuantumCircuit()
+        for qr in body.qregs:
+            ret.add_register(qr)
+        for cr in body.cregs:
+            ret.add_register(cr)
+        if len(ret.clbits) == 0:
+            ret.add_bits(body.clbits)
+        if len(ret.qubits) == 0:
+            ret.add_bits(body.qubits)
+
+        inlined_body = None
+        break_label = f'{loop_name}_end'
+        for index in indexset:
+            continue_label = f'{loop_name}_{index}'
+            inlined_body = self._inline_circuit(body,
+                                                continue_label,
+                                                break_label)
+            inlined_body = inlined_body.bind_parameters({loop_parameter: index})
+            ret.append(inlined_body,
+                       range(inlined_body.num_qubits),
+                       range(inlined_body.num_clbits))
+            ret.append(AerMark(continue_label, inlined_body.num_qubits),
+                       range(inlined_body.num_qubits), [])
+
+        if inlined_body:
+            ret.append(AerMark(break_label, inlined_body.num_qubits),
+                       range(inlined_body.num_qubits), [])
+        return ret
+
+    def _inline_while_loop_op(self, inst):
+        condition_tuple = inst.condition
+        body, = inst.params
+
+        self._last_flow_id += 1
```

Comment:
This variable shouldn't be stored in the class. If its a counter it should be passed as an argument and returned from the required functions.